doc/arch/adr-001-notify-integration.md

```diff
@@ -0,0 +1,424 @@
+   components and how they communicate to achieve specified functionality
+
+This document then goes through each of the concepts in the diagram and outline
+their role and responsibility.
+
+### ContentChange
+
+**Responsibility**: Representing a piece of content that is either new or has
+changed and should produce an alert.
+
+#### Shall
+
+- Store information such as:
+  - Title of content
+  - Description of content
+  - Change note (if any) associated with the update
+  - Path to content
+
+#### Should
+
+- Store information about what subscription criteria it matches (links, tags,
+  supertypes)
+- Have fields required for auditing
+  - Where the request came from
+  - Be able to determine whether and when it has been processed by a worker
+    (processed_at presence)
+- Be persisted forever
+
+#### May
+
+- In future, have an association to an entity that models the subscription
+  criteria information (links, tags, super types)
+- Be considered as part of an Alert supertype which may represent other types
+  of alerts in the system
+
+### Subscribable
+
+**Responsibility**: Being a type of list that a user can subscribe to and storing
+information specific to that list.
+
+#### Shall
+
+- Store information such as
+  - List name
+
+#### Should
+
+- Require that the list name is unique
+- Maintain a relationship with corresponding govdelivery list
+- Store information about what subscription criteria it matches
+  (links, tags, document_supertypes)
+- Have information to source where the Subscribable originated from
+
+#### May
+
+- In future, have an association to an entity that models the subscription
+  criteria information (links, tags, super types)
+- Release the requirement of a list name being unique if we are able to
+  establish a way to differentiate between lists
+- Store statistics related to the list
+- Be stored even after it is removed - for logging purposes, dependent on if
+  this will impact unique constraints
+
+#### To determine
+
+- In what scenarios is a Subscribable removed
+
+### Subscriber
+
+**Responsibility**: Represent an individual user that can subscribe to content,
+stores information related to all subscriptions the user may have
+
+#### Shall
+
+- Store information such as
+  - Email Address
+- Support an email address changing
+
+#### Should
+
+- Be deleteable, a user may wish to remove their account - this should not
+  break history of emails sent
+- Be used in the future for authentication purposes
+
+#### May
+
+- Store preferences that affect all of a users subscriptions
+- Have the ability to be disabled if there are problems with a users account
+- Store other personalisation information such as name
+- Be associated with statistics regarding the subscriber
+
+#### To determine
+- What should we allow in terms of removing a subscriber - do we use the
+  associated data as a log?
+
+### Subscription
+
+**Responsibility**: Store that a Subscriber has an association to a SubscriberList
```

Comment:
Should `SubscriberList` be `Subscribable`?

Reply:
yup 👍